config/module.go

```diff
@@ -213,3 +214,59 @@ func (m Module) EvaluateExePath(packagesDir string) (string, error) {
 	}
 	return m.ExePath, nil
 }
+
+// EvaluateFirstRunPath returns absolute FirstRunPath from one of three sources (in order of precedence):
+// 1. if there is a meta.json in the exe dir, use that, except in local non-tarball case.
+// 2. if this is a local tarball and there's a meta.json next to the tarball, use that.
+// 3. otherwise use the exe path from config, or fail if this is a local tarball.
+// Note: the working directory must be the unpacked tarball directory or local exec directory.
+func (m Module) EvaluateFirstRunPath(packagesDir string) (string, error) {
```

Comment:
Largely copypasta from `EvaluateExePath`, but allows the `first_run` script path to come from `meta.json` - we don't allow users to define the path in the module config.